xbmc/windowing/linux/OSScreenSaverFreedesktop.cpp

```diff
@@ -41,20 +54,10 @@ void COSScreenSaverFreedesktop::Inhibit()
     return;
   }
 
-  CDBusMessage inhibitMessage(SCREENSAVER_INTERFACE, SCREENSAVER_OBJECT, SCREENSAVER_INTERFACE, "Inhibit");
-  inhibitMessage.AppendArguments(KODI::LINUX::DESKTOP_FILE_NAME, g_localizeStrings.Get(14086));
-  if (!inhibitMessage.SendSession())
-  {
-    // DBus call failed
-    CLog::Log(LOGERROR, "Inhibiting freedesktop screen saver failed");
-    return;
-  }
-
-  if (!inhibitMessage.GetReplyArguments(&m_cookie))
-  {
-    CLog::Log(LOGERROR, "Could not retrieve cookie from org.freedesktop.ScreenSaver Inhibit response");
-    return;
-  }
+  m_proxy->callMethod("Inhibit")
+      .onInterface(SCREENSAVER_INTERFACE)
+      .withArguments(KODI::LINUX::DESKTOP_FILE_NAME, g_localizeStrings.Get(14086))
```

Comment:
CCompileInfo::GetAppName() usage instead of the desktop_file_name that you've PR'd to remove?

Reply:
lol you're right. I honestly didn't even realize as this is basically debased code from 2018